sdk/core/Azure.Core.TestFramework/src/ProxyTransport.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using Azure.Core.Tests.TestFramework;
+
+namespace Azure.Core.TestFramework
+{
+    public class ProxyTransport : HttpPipelineTransport
```

Comment:
Why do we have both the transport and the policy?

Reply:
The transport is needed so that the requestId can be set using Random right away. We could put all the logic in here, but the policy seems more appropriate as we are modifying the request, and we can also ensure the modification happens late.

Reply:
Also, proxy policy needs to set the recording ID for each test, whereas proxy transport is per test class